### PR TITLE
Spc events

### PR DESCRIPTION
**Pull request by lvdoorn**
The super punch card now has a legend, and uses real data.

### CODE REVIEW

File: src/aggregators/extendedPunchCardAggregator.js [hunk 1]
```diff
@@ -162,14 +162,19 @@ function ExtendedPunchCardAggregator(userName, platform, from, to) {
 
             var viewConcatinated = conversationViewEvents.concat(codeViewEvents).concat(commitViewEvents);
             var writeConcatinated = writeComments.concat(writeInlineComments);
-            var earliest = conversationViewEvents.sort(function (a, b) {
+            var sorted = conversationViewEvents.sort(function (a, b) {
```
Comment (mboom) [inline src/aggregators/extendedPunchCardAggregator.js]: If you use real data, then why you need this?

File: src/modules/super-punch-card.js [hunk 2]
```diff
@@ -80,10 +80,12 @@ define(function () {
...
     /*jshint ignore:end*/
```
Comment (mboom) [inline src/modules/super-punch-card.js]: If you use real data, then why you need this?

Comment (lvdoorn) [inline src/modules/super-punch-card.js]: This is just for the purpose of testing. You are correct, I will remove it.

File: src/modules/super-punch-card.js [hunk 3]
```diff
@@ -184,6 +109,25 @@ var fetching = false;
...
+
+    function createLegend() {
+        var res = [];
+        for (var i = 0; i < EVENT_TEXT.length; i++) {
+            var lText = EVENT_TEXT[i];
+            var lColor = color(i);
+            var lType = 'rect';
+            res.push({
+                "type": lType,
+                //"style": "stroke:rgb(51, 125, 212);stroke-width:2px;fill:rgba(51, 125, 212,0.5);",
```
Comment (mboom) [inline src/modules/super-punch-card.js]: Maybe you should remove this line?

File: src/modules/super-punch-card.js [hunk 4]
```diff
@@ -319,9 +264,60 @@ var fetching = false;
...
+                                .attr("y", startingY + 15)
+                                .attr("text-anchor", "start")
+                                .text(legendData[i].text);
+                            break;
+                    }
+                }
+            }
+
+
             //var prNumbers = getPrNumbers(data.sem_sessions); /*jshint ignore:line*/
```
Comment (mboom) [inline src/modules/super-punch-card.js]: also here

File: src/modules/super-punch-card.js [hunk 5]
```diff
@@ -319,9 +264,60 @@ var fetching = false;
...
-            var prNumbers = [7, 9];
-            var color = d3.scale.category10();
+            var prNumbers = [7, 9]; // todo
```
Comment (mboom) [inline src/modules/super-punch-card.js]: How do you mean: todo?

File: src/modules/super-punch-card.js [hunk 6]
```diff
@@ -409,7 +405,7 @@ var fetching = false;
             .attr("y", function (d) { return yScale(d) - rectHeight / 2; })
             .attr("x", xScale(0))
             .attr("height", rectHeight)
-            .attr("width", xScale(24)-xScale(0))
+            .attr("width", xScale(24) - xScale(0))
             //.attr("rx", 6)
             //.attr("ry", 6)
```
Comment (mboom) [inline src/modules/super-punch-card.js]: also here

File: src/modules/super-punch-card.js [hunk 7]
```diff
@@ -449,7 +445,7 @@ var fetching = false;
             })
             .style("fill", function (d) { return color(getColor(d)); });
             //.style("fill", function (d) { return c10(getPrNumber(d)); });
```
Comment (mboom) [inline src/modules/super-punch-card.js]: and here

File: src/modules/super-punch-card.js [hunk 8]
```diff
@@ -499,53 +495,40 @@ var fetching = false;
             .style("cursor", "pointer");*/
```
Comment (mboom) [inline src/modules/super-punch-card.js]: and here

File: src/modules/super-punch-card.js [hunk 9]
```diff
@@ -578,58 +560,70 @@ var fetching = false;
...
                 if (!map.get(daysAgo)) {
-                        new ExtendedPunchCardAggregator(globalUserName, 'GitHub', timespan.start, timespan.end)
-                            .then(function (a) {
-                                drawEverything(a);
-                                map.set(daysAgo, a);
-                                fetching = false;
+                    new ExtendedPunchCardAggregator(globalUserName, 'GitHub', timespan.start, timespan.end)
+                        .then(function (a) {
+                            //a = eventData; // comment this out for real data
+                            //a = huge;
```
Comment (mboom) [inline src/modules/super-punch-card.js]: Also here

File: src/modules/super-punch-card.js [hunk 10]
```diff
@@ -642,44 +636,40 @@ var fetching = false;
...
                     .attr("class", "visibleTicks noAxis removeAxis")
-                    .attr("transform", "translate(0," + (y + 30 * sessionNumbers.length) +")")
+                    .attr("transform", "translate(0," + (y + 70 + 30 * sessionNumbers.length) + ")")
                     .call(dayXAxis);
             }
 
             function drawViewEvents(vdata, alldata) {
                 var sessionNumbers = getSessionIdList(vdata);
 
-                //g.selectAll('#pr-bar').remove();
```
Comment (mboom) [inline src/modules/super-punch-card.js]: and here

File: src/modules/super-punch-card.js [hunk 11]
```diff
@@ -642,44 +636,40 @@ var fetching = false;
...
                 .attr('stroke-width', 20);
 
                 d3.select('#' + module.name).select('svg')
                     .transition()
                     .duration(300)
-                        .attr('viewBox', '0 0 1440 ' + (h + 30 * sessionNumbers.length + 10));
+                        .attr('viewBox', '0 0 1440 ' + (h + 60 * sessionNumbers.length + 50));
 
                 //var tip2 = d3.tip()
                 //    .attr('class', 'd3-tip')
```
Comment (mboom) [inline src/modules/super-punch-card.js]: and here

Comment (lvdoorn) [inline src/modules/super-punch-card.js]: This code should be used in the future to add tooltips to the newly drawn graphs.

Comment (mboom) [inline src/modules/super-punch-card.js]: I will help you with that.

**Pull request merged**